### ISSUE TITLE
REST API IP Filter doesn't work

### ISSUE THREAD

**Issue by maxim-smotrov** [ant-media/Ant-Media-Server#3638]
Hi! I need to use the REST API to delete the VODs when I don't need them anymore.
I added my IP address to the REST API IP Filter as described here: https://github.com/ant-media/Ant-Media-Server/wiki/REST-API-Security-Documentation
But I'm still getting the 403 "Not allowed IP". Below is the video of what I'm doing.
Please help me set up the IP filtering.

https://user-images.githubusercontent.com/12084862/134365456-4954d0ec-f007-4089-aec5-c955ef288254.mp4

**Comment by yashtandon113** [comment]
Hi @maxim-smotrov 
Please check this link --> https://github.com/ant-media/Ant-Media-Server/wiki/Frequently-Asked-Questions#how-to-enable-ip-filter-for-ant-media-servers-behind-load-balancer-in-aws
You have to add those additional lines in your server.
Please let me know if it works for you.

**Comment by maxim-smotrov** [comment]
@yashtandon113 thank you for the reply. I tried the steps from the link. Posted below the xml fragment I have now. But the IP filter still doesn't work. One thing to note is, I'm using Google Cloud Load Balancer, not Amazon. Can that be an issue?
```xml
<property name="valves">
        <list>
                <bean id="valve.access" class="org.apache.catalina.valves.RemoteIpValve" />
                <bean id="valve.access" class="org.apache.catalina.valves.AccessLogValve">
                        <property name="directory" value="log" />
                        <property name="prefix" value="${http.host}_access" />
                        <property name="suffix" value=".log" />
                        <property name="pattern" value="common" />
                        <property name="requestAttributesEnabled" value="true" />
                        <property name="rotatable" value="true" />
                </bean>

                <!-- cluster start -->
                <bean id="valve.cluster" class="io.antmedia.enterprise.cluster.HttpLiveStreamValve">
                        <property name="tokenGenerator" ref="tokenGenerator" />
                        <property name="asyncSupported" value="true" />
                </bean>
                <!-- cluster end -->
        </list>
</property>
```

**Comment by mekya** [comment]
Hi @maxim-smotrov ,

Sorry for late reply.  I think the problem is likely a kind of configuration issue in the Google Cloud Load Balancer.
Load Balancer should add the  [X-Forwarded-For](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For) header for the original IP address. 

If that is not a case, here is a workaround solution. You can just disable the REST IP Filtering and use [JWT REST Security](https://github.com/ant-media/Ant-Media-Server/wiki/JWT-REST-API-Filter). 

@muratugureminoglu , is it possible that we can make a test with Google Cloud Load Balancer to make sure about the problem?

**Comment by mekya** [comment]
TL;DR;
The IP Addresses reported by some tools such as whatsmyip.com may not be the one when accessing the server side. I think it is something on ISV side.  The solution is having static ip or adding ISV network in CIDR notation or using JWT token.

Hi @maxim-smotrov ,
Sorry for late reply, I've made several tests including AWS ELB and Google Cloud Load Balancer.  I've run an instance behind the load balancer and I've reproduced the problem. Here is the scenario

1. When I check my ip address on https://whatsmyip.com/ . It reports that it's 37.202.49.201

<img width="1508" alt="Screen Shot 2022-01-13 at 13 55 32" src="https://user-images.githubusercontent.com/3456251/149347134-86a94e4d-055d-4ca1-a67e-f46f6b9fd9f5.png">

2. I've added this IP address to the REST IP Filter and make a request from my computer
3. I've received 403 Forbidden issue as mentioned.
4. I've checked the access log and it shows that request is coming from 37.202.49.204 as highlighted in the image below
<img width="1179" alt="Screen Shot 2022-01-13 at 13 55 19" src="https://user-images.githubusercontent.com/3456251/149347751-1455ba0b-a654-4c99-b204-e118eb3338d6.png">
5. I've added 37.202.49.204 to the IP Filter and then it returns the response as shown below
<img width="1392" alt="Screen Shot 2022-01-13 at 15 32 35" src="https://user-images.githubusercontent.com/3456251/149348195-ca4485cf-435e-4a6b-8925-8818f69a186a.png">
 

Solution:
1. Use static IP. This is the case if you're running your own instance in the cloud
2. Guess ISV network and add it  to the IP Filter in CIDR notation such as 37.202.49.0/24
3.  Use JWT Token. I think this is the best and secure way if source IP addresses can be changed
